onnxruntime/core/providers/cpu/ml/ml_common.h

```diff
@@ -234,125 +195,111 @@ static inline void multiclass_probability(int64_t classcount, const std::vector<
   }
 }
 
-static const float ml_sqrt2 = 1.41421356f;
-
-static inline float ComputeLogistic(float val) {
-  float v = 1 / (1 + std::exp(-std::abs(val)));
+// y = \frac{1}{1+e^{-x}} , x \in R
+template <typename T>
+inline T ComputeLogistic(T val) {
+  T v = 1 / (1 + std::exp(-std::abs(val)));
   return (val < 0) ? (1 - v) : v;
 }
 
-static inline float ComputeProbit(float val) {
-  return ml_sqrt2 * ErfInv(2 * val - 1);
+// It assumes val is in [0,1]
+template <typename T>
+inline T ComputeProbit(T val) {
+  return static_cast<T>(M_SQRT2) * ErfInv(2 * val - 1);
 }
 
 static inline float sigmoid_probability(float score, float proba, float probb) {
   float val = score * proba + probb;
-  return 1 - ComputeLogistic(val);  // ref: https://github.com/arnaudsj/libsvm/blob/eaaefac5ebd32d0e07902e1ae740e038eaaf0826/svm.cpp#L1818
+  return 1 -
```

Comment:
I'm curious, you folded enums in one line but you unfolded this line into 3.

Reply:
The original code is too long, longer than 120 chars.